File: ImageViewer/app/src/main/java/com/example/imageviewer/ImageData.java
package com.example.imageviewer;

import android.net.Uri;

import java.io.Serializable;

public class ImageData implements Serializable
{
    private String name, stringUri;
    public static final String TAG = "image";

    public ImageData()
    {
        this.name = name;
        this.stringUri = stringUri;
    }

    public String getName() { return name; }

    public void setName(String name) {
        this.name = name;
    }

    public Uri getUri() {
        return Uri.parse(stringUri);
    }

    public void setUri(Uri uri) {
        this.stringUri =  uri.toString();
    }
}

File: app/src/main/java/com/lambdaschool/favoritepicturesgallery/ImageData.kt
package com.lambdaschool.favoritepicturesgallery

import android.net.Uri

import java.io.Serializable
//AH - Getting all the Image data in this class
class ImageData(fullPhotoUri: Uri) : Serializable {
    var name: String? = null
    var description: String? = null
    val fileUriString: String

    val fileUri: Uri
        get() = Uri.parse(fileUriString)

    init {
        this.fileUriString = fullPhotoUri.toString()
        val path = fullPhotoUri.path!!.split("/".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        this.name = path[path.size - 1]
        this.description = ""
    }
}
